node/src/command.rs

```diff
@@ -17,15 +17,74 @@ use sp_runtime::traits::{AccountIdConversion, Block as BlockT};
 use crate::{
 	chain_spec,
 	cli::{Cli, RelayChainCli, Subcommand},
-	service::{new_partial, ParachainNativeExecutor},
+	service::{new_partial, DevnetRuntimeExecutor, MainnetRuntimeExecutor},
 };
 
+/// Helper enum that is used for better distinction of different parachain/runtime configuration
+/// (it is based/calculated on ChainSpec's 'chain_spec' attribute)
+#[derive(Debug, PartialEq, Default)]
+enum Runtime {
+	/// This is the default runtime (actually based on rococo)
+	#[default]
+	Default,
+	Devnet,
+	Mainnet,
+}
+
+trait RuntimeResolver {
+	fn runtime(&self) -> Runtime;
+}
+/// Private helper that pattern matches on the input (which is expected to be a ChainSpec ID)
+/// and returns the Runtime accordingly.
+fn runtime(id: &str) -> Runtime {
+	if id.starts_with("dev") {
+		Runtime::Devnet
+	} else if id.starts_with("main") {
+		Runtime::Mainnet
+	} else {
+		log::warn!("No specific runtime was recognized for ChainSpec's Id: '{}', so Runtime::Default will be used", id);
+		Runtime::default()
+	}
+}
+/// Resolve runtime from ChainSpec ID
+impl RuntimeResolver for dyn ChainSpec {
+	fn runtime(&self) -> Runtime {
+		runtime(self.id())
+	}
+}
+/// Implementation, that can resolve [`Runtime`] from any json configuration file
+impl RuntimeResolver for PathBuf {
+	fn runtime(&self) -> Runtime {
+		#[derive(Debug, serde::Deserialize)]
+		struct EmptyChainSpecWithId {
+			id: String,
+		}
+
+		let file = std::fs::File::open(self).expect("Failed to open file");
+		let reader = std::io::BufReader::new(file);
+		let chain_spec: EmptyChainSpecWithId = sp_serializer::from_reader(reader)
+			.expect("Failed to read 'json' file with ChainSpec configuration");
+
+		runtime(&chain_spec.id)
+	}
+}
+
 fn load_spec(id: &str) -> std::result::Result<Box<dyn ChainSpec>, String> {
 	Ok(match id {
-		"dev" => Box::new(chain_spec::development_config()),
-		"template-rococo" => Box::new(chain_spec::local_testnet_config()),
-		"" | "local" => Box::new(chain_spec::local_testnet_config()),
-		path => Box::new(chain_spec::ChainSpec::from_json_file(std::path::PathBuf::from(path))?),
+		"dev" => Box::new(chain_spec::devnet::development_config()),
+		"template-rococo" => Box::new(chain_spec::devnet::local_testnet_config()),
```

Comment:
Shouldn't this be called `devnet-local` ?

Reply:
Yes it should. Adding `devnet-local` here.